src/tests/testClpMap.cc

```diff
@@ -211,7 +211,7 @@ TestClpMap::testMemoryLimit()
     // test that memory usage and entry count decrease when the map is shrinking
     // but prevent endless loops no matter how broken ClpMap implementation is
     auto iterationsLeft = m.entries();
-    CPPUNIT_ASSERT_MESSAGE("ClpMap::entries() returns bogus numbers",
+    CPPUNIT_ASSERT_MESSAGE("ClpMap::entries() returns reasonable numbers",
```

Comment:
The message should document the failure, see https://cppunit.sourceforge.net/doc/cvs/group___assertions.html#ga1

Reply:
I removed the controversial messages (commit cfea6fa) in hope to make progress. This fiasco also reminded me that it is actually a bad idea to use those messages in the first place. C++17 got it right when they changed the static_assert() API to stop requiring these messages: These messages are more trouble than they are worth, for several reasons! We have discussed this at length [before](https://github.com/squid-cache/squid/pull/785#discussion_r594365508).

> The message should document the failure

I disagree:

* #785 (i.e. commit 2bde9c7) has already established that assertion messages should document the condition. We spent hours on that decision. I support that decision, but if it was wrong, reversing it should be done in a dedicated PR, not here.
* CppUnit [documentation](https://cppunit.sourceforge.net/doc/cvs/group___assertions.html#ga1) and output do not contradict the stated principle IMO. AFAICT, CppUnit treats the message as supplementary output without _requiring_ it to describe either the condition or the failure. I did test CppUnit output before adding those messages, came to the conclusion that it looks OK (and even included a sample in PR commit 79714fc).
* All existing CPPUNIT_ASSERT_MESSAGE() calls in Squid document the condition. If nothing else, it is best to keep the code consistent (and change all messages at once if the decision to change messages is made in a dedicated PR).

Besides a link to the documentation and output (which I interpret in my favor), no _arguments_ have been presented to support the change AFAICT. IMHO, this ordeal is a good illustration why we should avoid those messages completely.

Reply:
I agree on avoiding the messages and am satisfied that the condition I was concerned is there.
Let's clear for merge and ship it

Reply:
> Let's clear for merge and ship it

Glad we found an acceptable solution! Please clear the PR for merging when you think it should be cleared.